docs/ibc/apps/ibcmodule.md

```diff
@@ -0,0 +1,350 @@
+  it supports.**
+  If there is no default version string for the application,
+  it should return an error if provided version is empty string.
+- `onChanOpenTry` will verify the relayer-chosen parameters along with the
+  counterparty-chosen version string and perform custom `TRY` logic.
+  If the relayer-chosen parameters
+  are invalid, the callback must return an error to abort the handshake.
+  If the counterparty-chosen version is not compatible with this modules
+  supported versions, the callback must return an error to abort the handshake.
+  If the versions are compatible, the try callback must select the final version
+  string and return it to core IBC.
+  `onChanOpenTry` may also perform custom initialization logic
+- `onChanOpenAck` will error if the counterparty selected version string
+  is invalid to abort the handshake. It may also perform custom ACK logic.
+
+Versions must be strings but can implement any versioning structure. If your application plans to
+have linear releases then semantic versioning is recommended. If your application plans to release
+various features in between major releases then it is advised to use the same versioning scheme
+as IBC. This versioning scheme specifies a version identifier and compatible feature set with
+that identifier. Valid version selection includes selecting a compatible version identifier with
+a subset of features supported by your application for that version. The struct is used for this
+scheme can be found in `03-connection/types`.
+
+Since the version type is a string, applications have the ability to do simple version verification
+via string matching or they can use the already impelemented versioning system and pass the proto
+encoded version into each handhshake call as necessary.
+
+ICS20 currently implements basic string matching with a single supported version.
+
+## Packet callbacks
+
+Just as IBC expected modules to implement callbacks for channel handshakes, IBC also expects modules to implement callbacks for handling the packet flow through a channel, as defined in the `IBCModule` interface.
+
+Once a module A and module B are connected to each other, relayers can start relaying packets and acknowledgements back and forth on the channel.
+
+![IBC packet flow diagram](https://ibcprotocol.org/_nuxt/img/packet_flow.1d89ee0.png)
+
+Briefly, a successful packet flow works as follows:
+
+1. module A sends a packet through the IBC module
+2. the packet is received by module B
+3. if module B writes an acknowledgement of the packet then module A will process the
+   acknowledgement
+4. if the packet is not successfully received before the timeout, then module A processes the
+   packet's timeout.
+
+### Sending packets
+
+Modules **do not send packets through callbacks**, since the modules initiate the action of sending packets to the IBC module, as opposed to other parts of the packet flow where msgs sent to the IBC
```

Comment:
```suggestion
Modules **do not send packets through callbacks**, since the modules initiate the action of sending packets to the IBC module, as opposed to other parts of the packet flow where messages sent to the IBC
```